lib/src/cli/commands/diff_command.dart

```diff
@@ -97,12 +101,36 @@ You may want to do this if you want to make sure
       defaultsTo: false,
       negatable: true,
     );
+    argParser.addOption(
+      _optionReportFormat,
+      help: 'Which output format should be used',
+      defaultsTo: 'cli',
+      allowed: ['cli', 'markdown', 'json'],
```

Comment:
I looked into the Documentation of the args package and couldn't find anything like enum options, of course I could use something like .toString() and parse it later into the enum, but is that the direction you were aiming for or your meant something else?

Reply:
Yes, you are right: Enums are not directly supported by the args package.
You can have a look at the DependencyCheckMode arg for mapping between enum and args:
https://github.com/bmw-tech/dart_apitool/blob/b64a5e773ae591e3ab7f54f79633bbc94d86f3f3/lib/src/cli/commands/diff_command.dart#L79

Reply:
I looked into the already existing tests, would you prefer extending the diff_command_test to check if the command handles the arguments accordingly? or what kind of Tests you had in mind? I was also looking into the unit test, but testing just the reporter class won't do much since it's just an abstract class which does nothing by its own.


So some feedback / opinion would be appreciated so I know what exactly to implement :) 


Wish yall a nice Friday :)

Reply:
Sorry for the late reply. Today has been busy.

I meant the different concrete Reporters and on a very high level.
I think it would be best to match the existing structure in the tests and add Unit Tests for the builders in a test/unit_test/diff/report directory.

If you want you additionally could add a few high level integration tests with concrete package versions but I think the unit tests are more important.

Reply:
@Leftwitch if you need any assistance or further clarification on this then just let me know

Reply:
Sorry for the late reply, yeah I was Just looking on how to write good unit Tests in Dart but did Progress slower then I personally hoped, I haven't done that in the past so It's a new skill for me to learn ^^, If you can assist that might be a good solution aswell then I can learn from the code ^^ 

Or I continue attempting and you review it when it's done?

I'm only talking about the tests here, the enum and rebase are done, I can push that if you want!

Reply:
Whatever suits you best. Either way is fine for me.

Reply:
Lets go with the first option so we make some progress after all!

Reply:
cool. Thanks.
Yes, I will add a few tests and push them to your branch. Not sure when exactly but I think should be either today or tomorrow

Reply:
@Leftwitch I added examples for an integration and a unit test (Markdown reporter and console reporter).
The unit tests basically only checks that nothing crashes and the integration test checks if the file has been produced and does some probing.
I think we should avoid going too deep into the produced content as this might change too frequently.
Once we find any issues we can concentrate on them and do explicit regression tests.

Reply:
Ah dang it, I didn't get a Notification for that one, weird, Anyways the Tests look good, I think I got the gist now of what you meant ^^

And also I agree, when we go too much Into detail the tests might become to flaky

Reply:
Also: Since the json format also generates a file we could add a test for that case aswell I think

Reply:
I added a Test-Case for the JSON-Report, it also attempts to parse the JSON and check for some essential properties

Reply:
@Leftwitch just ping me if you are done

Reply:
I added the Json test, if you don't have anything to add I'd say the PR is done so far :)